es-extension-examples/ext-example-vite/vite.config.js

```diff
@@ -0,0 +1,45 @@
+import reactRefresh from '@vitejs/plugin-react';
+import { defineConfig } from 'vite';
+import dotenv from "dotenv";
+dotenv.config();
+
+const urlPath = process.env.URLPATH || ''
+
+export default defineConfig({
+  /* when in lib mode, must define process.env */
+  define: {
+    'process.env': JSON.stringify({
+      NODE_ENV: process.env.NODE_ENV,
+    })
+  },
+  base: urlPath,
+  root: '.',
```

Comment:
The setting root: '.' tells Vite that entry files such as index.html are located in the base directory. In a monorepo which might run separate instances of Vite, this is crucial for creating boundaries.